client/src/components/FlightPhotos.vue

```diff
@@ -215,48 +222,67 @@ const photosChanged = () =>
   });
 
 // Show the file input dialog
-const onAddPhoto = () => photoInput.value.click();
+const onAddPhoto = () => photoInput.value?.click();
+interface HTMLInputEvent extends Event {
+  target: HTMLInputElement & EventTarget;
+}
 
 // Put selected photos in an upload cue and upload them
-const photoUploadQueue = ref([]);
-const onPhotoSelected = (event) => {
+
+interface UploadCueItem {
+  id: string;
+  photo: File;
+  status: "pending";
+}
+const photoUploadQueue = ref<UploadCueItem[]>([]);
+const onPhotoSelected = (e: Event) => {
+  const event = e as HTMLInputEvent;
+  if (!event.target.files?.length) return;
   // Check files count and substract alredy uploaded photos
   if (event.target.files.length > MAX_PHOTOS - _photos.value.length) {
     errorMessage.value = MAX_PHOTO_MESSAGE;
     return;
   }
   errorMessage.value = null;
-  [...event.target.files].slice(0, MAX_PHOTOS).forEach((photo) => {
+  // @ts-expect-error
+  const array = [...event.target.files];
+  array.slice(0, MAX_PHOTOS).forEach((photo) => {
     photoUploadQueue.value.push({
       id: uuidv4(),
       photo: photo,
       status: "pending",
     });
-    uploadPhoto(last(photoUploadQueue.value));
+    const lastItem = last(photoUploadQueue.value);
+    if (!lastItem) return;
+    uploadPhoto(lastItem);
   });
 };
 
-const uploadPhoto = async (item, { retryIndex = null } = {}) => {
-  let index = null;
+const uploadPhoto = async (item: UploadCueItem, retryIndex?: number) => {
+  let index: number | undefined = undefined;
   // Skip if this is a retry attempt
-  if (retryIndex === null) {
+  if (!retryIndex) {
     // Create a photo object and add it to the local photos list
     index = _photos.value.length;
-    const _phototmp = { id: null, description: "", uploadCue: item.id };
+    const _phototmp: NewPhoto = {
+      id: undefined,
+      description: "",
+      uploadCue: item.id,
+    };
     _photos.value[index] = _phototmp;
   } else {
     index = retryIndex;
 
     // Reset photo status
-    _photos.value[index].id = null;
+    _photos.value[index].id = undefined;
   }
-
   // Upload it
   try {
+    if (!getUserId.value || isNaN(index) || !props.flightId) return;
     const formData = new FormData();
     formData.append("image", item.photo, item.photo.name);
     formData.append("flightId", props.flightId);
-    formData.append("userId", getUserId);
+    formData.append("userId", getUserId.value); // Why did this work withput value?
```

Comment:
Gute Frage... aus der Doku:
value

    The field's value. This can be a string or [Blob](https://developer.mozilla.org/en-US/docs/Web/API/Blob) (including subclasses such as [File](https://developer.mozilla.org/en-US/docs/Web/API/File)). If none of these are specified the value is converted to a string.

Ich sehe jetzt nicht wie er das zu einem String umwandelt, aber meine Idee wäre jetzt das die Ref Klasse hier eine Methode anbietet, die dann das interne value herauszieht.

Kommentar kann dann weg?

Reply:
schon weg